scripts/install.js

```diff
@@ -109,18 +109,68 @@ function checkAndDownloadBinary() {
       return;
     }
 
-    download(sass.getBinaryUrl(), sass.getBinaryPath(), function(err) {
-      if (err) {
-        console.error(err);
-        return;
-      }
+    var tmpPath = getTempPath(sass.getBinaryName());
+    if (!(sass.hasBinary(tmpPath))) { // download and install
```

Comment:
You're both correct. The binaries won't class on different node/arch/os
combinations. However because we're no longer within node_modules changing
versions of node-sass would be an issue. When changing versions of
node-sass the installer would see a binary for their environment on disk
but it would not the correct binary for their node-sass version.

You'll need to create a folder for the node-sass version for this to work
as intended.
On 27 May 2016 9:21 PM, "Michael Riehemann" <notifications@github.com>
wrote:

> In scripts/install.js
> <https://github.com/sass/node-sass/pull/1566#discussion_r64891169>:
>
> > @@ -109,18 +109,68 @@ function checkAndDownloadBinary() {
> >        return;
> >      }
> >
> > -    download(sass.getBinaryUrl(), sass.getBinaryPath(), function(err) {
> > -      if (err) {
> > -        console.error(err);
> > -        return;
> > -      }
> > +    var tmpPath = getTempPath(sass.getBinaryName());
> > +    if (!(sass.hasBinary(tmpPath))) { // download and install
>
> The name of the binary file in the tmp folder based on
> sass.getBinaryName(). E.g. win32-x64-47_binding.node. If you use another
> version of node-sass, the name will change too.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sass/node-sass/pull/1566/files/1000858d85ae058e2d35093ea1a979b4742054b2#r64891169>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAjZWPgY_Jkj3Jfl8lXEHgVhkK4sTfQOks5qFtOugaJpZM4Ing0m>
> .
>